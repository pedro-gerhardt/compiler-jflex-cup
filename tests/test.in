class NomeDaClasse {
    int attr1;
    bool attr2;

    void func(int param, bool param2) { 
        teste = not (True and False); 
        return teste(1 + 3, True == False) + 1; 
        func(1 + 1, not True); 
        func(1 + 1, 1 == 2); 
        if (False) teste = 1;
        else return 2 + 33;
    }
}