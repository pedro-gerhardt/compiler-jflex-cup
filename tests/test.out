|--[class decl]
   |--[class]
   |--[NomeDaClasse]
   |--[{]
   |--[attr decl list]
      |--[attr_decl]
         |--[int]
         |--[attr1]
         |--[;]
      |--[attr_decl]
         |--[bool]
         |--[attr2]
         |--[;]
   |--[method decl]
      |--[void]
      |--[func]
      |--[(]
      |--[method decl args]
         |--[method decl val]
            |--[int]
            |--[param]
         |--[method decl args]
            |--[,]
            |--[method decl val]
               |--[bool]
               |--[param2]
            |--[]
      |--[)]
      |--[{]
      |--[command list]
         |--[not if command]
            |--[assign]
               |--[teste]
               |--[=]
               |--[not]
                  |--[bool expr]
                     |--[(]
                     |--[[and]]
                        |--[True]
                        |--[False]
                     |--[)]
            |--[;]
         |--[command list]
            |--[not if command]
               |--[return]
                  |--[+]
                     |--[method call]
                        |--[teste]
                        |--[(]
                        |--[method call args]
                           |--[+]
                              |--[1]
                              |--[3]
                           |--[method call args]
                              |--[,]
                              |--[[==]]
                                 |--[True]
                                 |--[False]
                              |--[]
                        |--[)]
                     |--[1]
               |--[;]
            |--[command list]
               |--[not if command]
                  |--[method call]
                     |--[func]
                     |--[(]
                     |--[method call args]
                        |--[+]
                           |--[1]
                           |--[1]
                        |--[method call args]
                           |--[,]
                           |--[not]
                              |--[True]
                           |--[]
                     |--[)]
                  |--[;]
               |--[command list]
                  |--[not if command]
                     |--[method call]
                        |--[func]
                        |--[(]
                        |--[method call args]
                           |--[+]
                              |--[1]
                              |--[1]
                           |--[method call args]
                              |--[,]
                              |--[[==]]
                                 |--[1]
                                 |--[2]
                              |--[]
                        |--[)]
                     |--[;]
                  |--[if]
                     |--[(]
                     |--[False]
                     |--[)]
                     |--[not if command]
                        |--[assign]
                           |--[teste]
                           |--[=]
                           |--[1]
                        |--[;]
                     |--[else]
                        |--[not if command]
                           |--[return]
                              |--[+]
                                 |--[2]
                                 |--[33]
                           |--[;]
      |--[}]
   |--[}]
Successful compilation!
